engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -880,6 +883,20 @@ private void execute() throws YarnException, IOException
         if (0 != exitStatus) {
           if (allocatedContainer != null) {
             numFailedContainers.incrementAndGet();
+            if (exitStatus != 1) {
+              // If container failure due to framework
+              AtomicInteger failedTimes = new AtomicInteger(1);
+              String hostname = allocatedContainer.container.getNodeId().getHost();
+              if (failedContainersMap.containsKey(hostname)) {
+                failedTimes = failedContainersMap.get(hostname);
+                int failureOnNode = failedTimes.incrementAndGet();
```

Comment:
Should the hostname not be removed from failedContainerMap once the container is allocated and works fine.. I mean what happens if during first 4 tries the container fails but in 5th try container works on host X.. I dont' see it getting removed from the failedContainerMap